### PR TITLE
fix(rpc/types): add pending transaction receipt types

### PR DESCRIPTION
**Pull request by kkovaacs**
These were marked required in version 0.29.0 of the specification.

### CODE REVIEW

**Comment by kkovaacs** [comment]
We are, I think. Can just wait and see how that update works out -- I expect that change to introduce separate receipt types for pending, but that's just pure speculation. We don't support returning receipts for pending transactions now, though, so I think this change is correct as it is for now.

**Review by CHr15F0x — APPROVED** [review]

**Comment by Mirko-von-Leipzig** [comment]
> We are, I think. Can just wait and see how that update works out -- I expect that change to introduce separate receipt types for pending, but that's just pure speculation. We don't support returning receipts for pending transactions now, though, so I think this change is correct as it is for now.

I've added pending transactions in my PR 😓 but can always just remove it until the dust settles.

**Comment by kkovaacs** [comment]
> I've added pending transactions in my PR sweat but can always just remove it until the dust settles.

Ok, in that case I think it's easier to just wait with this PR until we find out what the correct representation is for pending receipts in the spec.

**Review by CHr15F0x — APPROVED** [review]

File: crates/pathfinder/src/rpc.rs [hunk 1]
```diff
@@ -642,6 +642,7 @@ mod tests {
         ];
 
         let transaction_receipts = vec![
+            // FIXME: declare transactions don't have events, so this is invalid test data!
```
Comment (kkovaacs) [inline crates/pathfinder/src/rpc.rs:645]: This should be fixed before merging this I think.

I mean this was true even before this PR, but now we try to explicitly test the `getTransactionReceipt` RPC result which exposes the receipts.

File: crates/pathfinder/src/rpc.rs [hunk 2]
```diff
@@ -1537,6 +1538,13 @@ mod tests {
                     .unwrap();
                 // Only asserting the hash because translating from Sequencer receipt to RPC receipt is pita.
                 assert_eq!(receipt.hash(), expected.transaction_hash);
+                assert_matches!(
+                    receipt,
+                    // FIXME: this should test an invoke instead
```
Comment (kkovaacs) [inline crates/pathfinder/src/rpc.rs:1543]: And this one should be updated too after the test data has been fixed -- in that case we can actually assert on something meaningful inside the receipt.

**Review by Mirko-von-Leipzig — APPROVED** [review]

**Pull request merged**